src/sources/exec.rs

```diff
@@ -0,0 +1,886 @@
+            line,
+            &data_stream,
+            pid,
+            exit_status,
+            &exec_duration_millis,
+        );
+        let log = event.into_log();
+
+        assert_eq!(log[log_schema().host_key()], "Some.Machine".into());
+        assert_eq!(log[STREAM_KEY], STDOUT.into());
+        assert_eq!(log[PID_KEY], (8888_i64).into());
+        assert_eq!(log[COMMAND_KEY], config.command.into());
+        assert_eq!(log[log_schema().message_key()], "hello world".into());
+        assert_eq!(log[log_schema().source_type_key()], "exec".into());
+        assert_ne!(log[log_schema().timestamp_key()], "".into());
+    }
+
+    #[test]
+    fn test_streaming_create_event() {
+        let config = standard_streaming_test_config();
+        let hostname = Some("Some.Machine".to_string());
+        let line = Bytes::from("hello world");
+        let data_stream = Some(STDOUT.to_string());
+        let pid = Some(8888_u32);
+        let exit_status = None;
+        let exec_duration_millis = None;
+
+        let event = create_event(
+            &config,
+            &hostname,
+            line,
+            &data_stream,
+            pid,
+            exit_status,
+            &exec_duration_millis,
+        );
+        let log = event.into_log();
+
+        assert_eq!(log[log_schema().host_key()], "Some.Machine".into());
+        assert_eq!(log[STREAM_KEY], STDOUT.into());
+        assert_eq!(log[PID_KEY], (8888_i64).into());
+        assert_eq!(log[COMMAND_KEY], config.command.into());
+        assert_eq!(log[log_schema().message_key()], "hello world".into());
+        assert_eq!(log[log_schema().source_type_key()], "exec".into());
+        assert_ne!(log[log_schema().timestamp_key()], "".into());
+    }
+
+    #[test]
+    fn test_build_command() {
+        let config = ExecConfig {
+            mode: Mode::Streaming,
+            scheduled: None,
+            streaming: Some(StreamingConfig {
+                respawn_on_exit: default_respawn_on_exit(),
+                respawn_interval_secs: default_respawn_interval_secs(),
+            }),
+            command: vec!["./runner".to_owned(), "arg1".to_owned(), "arg2".to_owned()],
+            working_directory: Some(PathBuf::from("/tmp")),
+            include_stderr: default_include_stderr(),
+            event_per_line: default_events_per_line(),
+            maximum_buffer_size_bytes: default_maximum_buffer_size(),
+        };
+
+        let command = build_command(&config);
+
+        let mut expected_command = Command::new("./runner");
+        expected_command.kill_on_drop(true);
+        expected_command.current_dir("/tmp");
+        expected_command.args(vec!["arg1".to_owned(), "arg2".to_owned()]);
+
+        // Unfortunately the current_dir is not included in the formatted string
+        let expected_command_string = format!("{:?}", expected_command);
+        let command_string = format!("{:?}", command);
+
+        assert_eq!(expected_command_string, command_string);
```

Comment:
I'm not super crazy about comparing debug display representations, but I see that `Command` doesn't implement `PartialEq` or `Eq`.

Reply:
Agreed (hopefully it will not turn into a flakey test)